hyperspy/signal.py

```diff
@@ -4710,36 +4740,60 @@ def map(
         parameter is variable:
 
         >>> im = hs.signals.Signal2D(np.random.random((10, 64, 64)))
-        >>> sigmas = hs.signals.BaseSignal(np.linspace(2,5,10)).T
+        >>> sigmas = hs.signals.BaseSignal(np.linspace(2, 5, 10)).T
         >>> im.map(scipy.ndimage.gaussian_filter, sigma=sigmas)
 
+        Rotate the two signal dimensions, with different amount as a function
+        of navigation index. Delay the calculation by getting the result
+        lazily. The calculation is then done using the compute method.
+
+        >>> from scipy.ndimage import rotate
+        >>> s = hs.signals.Signal2D(np.random.random((5, 4, 40, 40)))
+        >>> s_angle = hs.signals.BaseSignal(np.linspace(0, 90, 20).reshape(5, 4)).T
+        >>> s.map(rotate, angle=s_angle, reshape=False, lazy_result=True)
+        >>> s.compute()
+
+        Rotate the two signal dimensions, with different amount as a function
+        of navigation index. In addition, the result is returned as a new
+        signal, instead of replacing the old signal.
+
+        >>> s = hs.signals.Signal2D(np.random.random((5, 4, 40, 40)))
+        >>> s_angle = hs.signals.BaseSignal(np.linspace(0, 90, 20).reshape(5, 4)).T
+        >>> s_rot = s.map(rotate, angle=s_angle, reshape=False, inplace=True)
+
         Note
         ----
         Currently requires a uniform axis.
 
         """
-        if self.axes_manager.navigation_shape == () and self._lazy:
-            _logger.info("Converting signal to a non-lazy signal because there are no nav dimensions")
-            self.compute()
-
-        # Sepate ndkwargs depending on if they are BaseSignals.
+        if lazy_result is None:
+            lazy_result = self._lazy
+        if ragged is None:
+            ragged = self.ragged
```

Comment:
For `ragged=None`, this is not what the docstring describe?
The current behaviour is defined by:
https://github.com/hyperspy/hyperspy/blob/a2e787185743ec1dbae990d67715f2cf828879dc/hyperspy/signal.py#L4978-L4979

Reply:
I rewrote the docstring text for the ragged parameter to reflect the changes.

The bigger potential issue is that this breaks the current behavior for `s.map` of non-lazy signals. In the current `RELEASE_next_minor` the `_map_iterate` automatically turns the signal into a ragged signal, if the returned signal dimensions has different shapes. However, it does this by effectively doubling the required memory in the process.

Lets assume an input navigation shape of the signal as `(128, 128)`. It does this by saving the results to a `np.empty((128, 128), dtype=object)`, then checking how the shapes looks like in each navigation position. If they all have the same size, the array is converted to a non-`object` array via:

```python
res_data = np.stack(res_data.ravel()).reshape(shape)
```

However, (at least on my computer) this doubles the memory requirement in this process. For big datasets, this can be an issue.

----------

This same functionality could easily be replicated in the new `s.map` in this pull request, but I think the "double memory need" is really bad. Essentially, to handle ragged results automatically we either have to find a way to "convert" an `object` NumPy array into a "standard" array (ideally without using a lot of memory, OR possibly check several navigation positions to see of the returned output is always the same.

This code tries to make a new "non-object" array from an `object` array, but doubles up the memory use.

```python
import numpy as np
data = np.empty((128, 128), dtype=object)
for iy, ix in np.ndindex(data.shape):
    data[iy, ix] = np.random.randint(0, 1024, size=(256, 256), dtype=np.uint16)
data_new = np.empty((128, 128, 256, 256), dtype=np.uint16)
for iy, ix in np.ndindex(data.shape):
    data_new[iy, ix] = data[iy, ix]
```

Reply:
Yes, the docstring simply needed to be updated! I wasn't not a big fan of this `None` option.